src/sparse_blas/backends/mkl_common/mkl_spmv.cxx

```diff
@@ -191,34 +200,36 @@ sycl::event internal_spmv(sycl::queue& queue, oneapi::mkl::transpose opA, const
         auto x_usm = x_handle->get_usm_ptr<T>();
         auto y_usm = y_handle->get_usm_ptr<T>();
         if (A_view.type_view == matrix_descr::triangular) {
-            return oneapi::mkl::sparse::trmv(queue, A_view.uplo_view, opA, A_view.diag_view,
-                                             host_alpha, backend_handle, x_usm, host_beta, y_usm,
-                                             dependencies);
+            RETHROW_ONEMKL_EXCEPTIONS_RET(
```

Comment:
Well I wish this could have been a templated function but it was too tricky to get it working with functions that return an event and functions that don't unfortunately. Glad you like it though!